### PR TITLE
Prevent regex queries from using case insensitive index and fix explain bug

### PR DESCRIPTION
**Pull request by pocketcolin**
### Issue
After the release of Parse Server 4.0.0, we noticed a significant performance decrease when running regex queries against `Username` or `Email` on the `_User` class. After much research and conversation with the Mongo community, we determined that the issue was related to how Mongo determines which index to use and it was sometimes picking the new case insensitive index (released in Parse Server 4.0). As that new index is a collation and regex queries are not able to utilize collation indexes, this caused performance issues in large collections.

Our initial report and further discussion can be found here: https://github.com/parse-community/parse-server/issues/6559

### Solution
I've updated the `_rawFind()` method to use the appropriate case sensitive (i.e., non-collation) index only if the following are true:
1. The query includes a regex operator on either of the `username` or `email` fields AND a custom index hint was not passed.
2. Both of the automatically generated indexes (`case_insensitive_{fieldName}` and `{fieldName}_1`) exist.

This solution will prevent the case insensitive index from being used for regex queries on username or email fields in the `_User` class while still allowing users to bypass this system if they want.

### Explain fix
While working on this fix, I also noticed that the newly implemented `.explain()` method throws an error if you try to run it on a `_User` query due to the fact that the result parser expects an array of results, but Mongo returns an object when the explain is passed. This PR also includes a fix for that issue.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,3 +1,4 @@
 semi: true
 trailingComma: "es5"
 singleQuote: true
+arrowParens: "avoid"
```
Comment (pocketcolin) [inline .prettierrc:4]: Prettier was updated in 4.2.0 and that update changed the default value of `arrowParens`. This just resets that default back to what it was prior to the Prettier update.

File: src/RestQuery.js [hunk 2]
```diff
@@ -188,7 +188,7 @@ function RestQuery(
 // Returns a promise for the response - an object with optional keys
 // 'results' and 'count'.
 // TODO: consolidate the replaceX functions
-RestQuery.prototype.execute = function(executeOptions) {
+RestQuery.prototype.execute = function (executeOptions) {
```
Comment (pocketcolin) [inline src/RestQuery.js:191]: Just a quick note here: All of these function parenthesis space additions are a result of the Prettier update as well. Prettier's most recent release includes a change to function parenthesis spacing that cannot be disabled. 

Full discussion here: https://github.com/prettier/prettier/pull/3903

**Comment by dplewis** [comment]
@pocketcolin Thanks for the PR and sorry for the late reply. Can you rebase from master? This looks good.

**Comment by pocketcolin** [comment]
@dplewis I think this PR can probably just be closed at this point. A much larger conversation around this issue happened [here](https://github.com/parse-community/parse-server/issues/6559), but it was eventually closed by the stale bot. Happy to continue that conversation if you think this is something that Parse could really benefit from.

**Pull request closed without merging**